pkg/controllers/deliverable_reconciler.go

```diff
@@ -110,34 +110,34 @@ func (r *DeliverableReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	serviceAccountName, serviceAccountNS := getServiceAccountNameAndNamespaceForDeliverable(deliverable, delivery)
 
-	secret, err := r.Repo.GetServiceAccountSecret(ctx, serviceAccountName, serviceAccountNS)
-	if err != nil {
-		r.conditionManager.AddPositive(conditions.ServiceAccountSecretNotFoundCondition(err))
-		return r.completeReconciliation(ctx, deliverable, nil, fmt.Errorf("failed to get secret for service account [%s]: %w", fmt.Sprintf("%s/%s", serviceAccountNS, serviceAccountName), err))
+	secret, getSecretErr := r.Repo.GetServiceAccountSecret(ctx, serviceAccountName, serviceAccountNS)
+	if getSecretErr != nil {
+		r.conditionManager.AddPositive(conditions.ServiceAccountSecretNotFoundCondition(getSecretErr))
+		return r.completeReconciliation(ctx, deliverable, nil, fmt.Errorf("failed to get secret for service account [%s]: %w", fmt.Sprintf("%s/%s", serviceAccountNS, serviceAccountName), getSecretErr))
 	}
 
-	resourceRealizer, err := r.ResourceRealizerBuilder(secret, deliverable, deliverable.Spec.Params, r.Repo, delivery.Spec.Params, buildDeliverableResourceLabeler(deliverable, delivery))
+	resourceRealizer, buildRealiserErr := r.ResourceRealizerBuilder(secret, deliverable, deliverable.Spec.Params, r.Repo, delivery.Spec.Params, buildDeliverableResourceLabeler(deliverable, delivery))
 
-	if err != nil {
-		r.conditionManager.AddPositive(conditions.ResourceRealizerBuilderErrorCondition(err))
-		return r.completeReconciliation(ctx, deliverable, nil, cerrors.NewUnhandledError(fmt.Errorf("failed to build resource realizer: %w", err)))
+	if buildRealiserErr != nil {
+		r.conditionManager.AddPositive(conditions.ResourceRealizerBuilderErrorCondition(buildRealiserErr))
+		return r.completeReconciliation(ctx, deliverable, nil, cerrors.NewUnhandledError(fmt.Errorf("failed to build resource realizer: %w", buildRealiserErr)))
 	}
 
 	resourceStatuses := statuses.NewResourceStatuses(deliverable.Status.Resources, conditions.AddConditionForResourceSubmittedDeliverable)
-	err = r.Realizer.Realize(ctx, resourceRealizer, delivery.Name, realizer.MakeDeliveryOwnerResources(delivery), resourceStatuses)
 
-	if err != nil {
-		conditions.AddConditionForResourceSubmittedDeliverable(&r.conditionManager, true, err)
-	} else {
-		r.conditionManager.AddPositive(conditions.ResourcesSubmittedCondition(true))
-	}
+	var reconcileErr error
```

Comment:
when first looking at the code I noticed that errors above this line seems to not be "aggregated" into `reconcileErr`, so we could have them as plain `err` - wdyt of that?

I think it's indeed worth having specialized naming for the errors after here, but not sure about the ones before (although I can see the argument for consistency 🤔 )

Reply:
We were, by mistake, logging err instead of trackingErr. I believe this occurred because there are so many error return values in this code block that the issue did not stand out during a refactor/shift of code (or the subsequent review).

I'd like to avoid shadowing `err` here, for now,  until we find an abstraction where the ONLY kind of err return value in this block, is one we use or ignore.